builtin/k8s/task.go

```diff
@@ -64,6 +64,12 @@ type TaskLauncherConfig struct {
 
 	// The namespace to use for launching this task in Kubernetes
 	Namespace string `hcl:"namespace,optional"`
+
+	// Optionally define various cpu resource limits and requests for kubernetes pod containers
+	CPU *ResourceConfig `hcl:"cpu,block,optional"`
+
+	// Optionally define various memory resource limits and requests for kubernetes pod containers
+	Memory *ResourceConfig `hcl:"memory,block,optional"`
```

Comment:
So I think this is wrong, if we have a `block` it can be `optional` if the struct fields are also optional. Otherwise I think this is a parse error and we want this instead:

```suggestion
	Memory *ResourceConfig `hcl:"memory,block"`
```

Same with `CPU` too.

```
2022-05-04T15:20:49.722Z [DEBUG] waypoint.runner.agent.runner: plugin successfully launched and connected: job_id=01G27TG44Y9EVXBS6N6BPMMBFW job_op=*gen.Job_StartTask
2022-05-04T15:20:49.722Z [DEBUG] waypoint.runner.agent.runner: decoding task plugin config: job_id=01G27TG44Y9EVXBS6N6BPMMBFW job_op=*gen.Job_StartTask
panic: invalid hcl field tag kind "block,optional" on *struct { Requested string "hcl:\"request,optional\""; Limit string "hcl:\"limit,optional\"" } "CPU"

goroutine 54 [running]:
github.com/hashicorp/hcl/v2/gohcl.getFieldTags({0x3a07760, 0xc000a791a0})
        /go/pkg/mod/github.com/hashicorp/hcl/v2@v2.10.1-0.20210621220818-327f3ce2570e/gohcl/schema.go:176 +0x625
...
...
```

Reply:
Ah, gotcha. I had only added this because it seems like the docs site keys off of `optional` to determine whether or not the parameter shows up in the required parameters field or not. Is there a way for the auto gen docs to place these parameters into the optional section instead of marking them both as required?

Reply:
Also, all blocks are optional.